------------------------------------------------------------------------
r21857 | gd | 2007-03-16 17:20:47 +0100 (Fri, 16 Mar 2007) | 4 lines

Stop pretending to be Vista in the %a macro towards Samba clients.

Guenther

------------------------------------------------------------------------
Index: source/smbd/negprot.c
===================================================================
--- source/smbd/negprot.c	(revision 21856)
+++ source/smbd/negprot.c	(revision 21857)
@@ -259,7 +259,9 @@ static int reply_nt1(char *inbuf, char *
 	if ( (SVAL(inbuf, smb_flg2) & FLAGS2_EXTENDED_SECURITY) &&
 		((SVAL(inbuf, smb_flg2) & FLAGS2_UNKNOWN_BIT4) == 0) ) 
 	{
-		set_remote_arch( RA_VISTA );		
+		if (get_remote_arch() != RA_SAMBA) {
+			set_remote_arch( RA_VISTA );
+		}
 	}
 
 	/* do spnego in user level security if the client
